modules/govuk_jenkins/templates/jobs/search_api_learn_to_rank.yaml.erb

```diff
@@ -0,0 +1,89 @@
+---
+- scm:
+    name: search-api
+    scm:
+      - git:
+          url: git@github.com:alphagov/search-api.git
+          branches:
+            - migrate-ltr-to-jenkins
+- job:
+    name: search-api-learn-to-rank
+    display-name: Search API Learn to Rank
+    project-type: freestyle
+    description: This job triggers Search API to generate training data, then use AWS SageMaker to train and deploy a model.
+    scm:
+      - search-api
+    logrotate:
+        numToKeep: 100
+    triggers:
+      - timed: |
+          TZ=Europe/London
+          0 22 * * SUN
+    properties:
+      - build-discarder:
+          days-to-keep: 30
+          artifact-num-to-keep: 5
+    publishers:
+      - trigger-parameterized-builds:
+          - project: Success_Passive_Check
+            condition: SUCCESS
+            predefined-parameters: |
+                NSCA_CHECK_DESCRIPTION=<%= @service_description %>
+                NSCA_OUTPUT=<%= @service_description %> success
+          - project: Failure_Passive_Check
+            condition: FAILED
+            predefined-parameters: |
+                NSCA_CHECK_DESCRIPTION=<%= @service_description %>
+                NSCA_OUTPUT=<%= @service_description %> failed
+    parameters:
+      - string:
+          name: AWS_ROLE_ARN
+          description: The ARN for the AWS Role to assume
+          default: <%= @aws_role_arn %>
+      - string:
+          name: ELASTICSEARCH_URI
+          description: The URI of Elasticsearch cluster for generating training data
+          default: <%= @elasticsearch_uri %>
+      - string:
+          name: IMAGE
+          description: The image name to use in SageMaker for training the model
+          default: <%= @image %>
+      - string:
+          name: TRAIN_INSTANCE_TYPE
+          description: The type of instance used to train the model
+          default: <%= @train_instance_type %>
+      - string:
+          name: TRAIN_INSTANCE_COUNT
+          description: The number instances used to train the model
+          default: <%= @train_instance_count %>
+      - string:
+          name: DEPLOY_INSTANCE_TYPE
+          description: The type of instance used to host the deployed model
+          default: <%= @deploy_instance_type %>
+      - string:
+          name: DEPLOY_INSTANCE_COUNT
+          description: The number instances used to host the deployed model
+          default: <%= @deploy_instance_count %>
+      - string:
+          name: GOVUK_ENVIRONMENT
+          default: <%= @environment %>
+    builders:
+      - shell: |
+          set -ex
+          chmod +x ./ltr/jenkins/start.sh
+          ./ltr/jenkins/start.sh
+    wrappers:
+      - ansicolor:
+          colormap: xterm
+      - credentials-binding:
+          - ssh-user-private-key:
+              credential-id: learn-to-rank-ssh-key 
+              key-file-variable: SSH_PRIVATE_KEY_PATH
+              username-variable: SSH_USERNAME
```

Comment:
Forgive me, how does this part work? I see we have piped in a variable `ssh_private_key` to the puppet class, but I'm not sure of the lineage of `SSH_PRIVATE_KEY_PATH` (though I see this env var used in your search-api branch) or `SSH_USERNAME`.

Reply:
Ah `ssh_private_key` puppet variable was mistake. Now, the SSH key is added to the Jenkins credentials manager and this credential binding provides `SSH_PRIVATE_KEY_PATH ` variable which is a file path to the SSH key.

Previously, SSH key was injected direct into an env var, then the `ltr/jenkins/start.sh` script saved it to a file to be used by the SSH command. Using the credential binding and manager seemed neater - but realised we can't use puppet to add creds to the Jenkins credential manager (I'd manually added it before).

So thinking now will switch back to the injected env var and using the puppet `ssh_private_key` to pull the secret from hieradata.

Reply:
Had another look - turns out when injecting the credential into a env var replaces the `\n` with spaces, which is what ultimately was the problem and broke the key. I've got it working now, so that it uses the SSH_PRIVATE_KEY env var and we can save the key in govuk-secrets. Had to add a line to replace the spaces with new lines.